gateway/sds_gateway/visualizations/api_views.py

```diff
@@ -0,0 +1,485 @@
+                metadata={
+                    "requested_by": str(request.user.uuid),
+                    "requested_at": request.data.get("timestamp"),
+                    "image_dimensions": request.data.get("dimensions", {}),
+                },
+            )
+
+            # Start spectrogram processing
+            # This will use the cog pipeline to generate the spectrogram
+            self._start_spectrogram_processing(spectrogram_data)
+
+            serializer = PostProcessedDataSerializer(spectrogram_data)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        except Exception as e:  # noqa: BLE001
+            log.error(f"Error creating spectrogram: {e}")
+            return Response(
+                {"error": "Failed to create spectrogram"},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+    @extend_schema(
+        summary="Get spectrogram status",
+        description="Get the status of a spectrogram generation job",
+        parameters=[
+            OpenApiParameter(
+                name="capture_uuid",
+                type=str,
+                location=OpenApiParameter.PATH,
+                description="UUID of the capture",
+            ),
+            OpenApiParameter(
+                name="job_id",
+                type=str,
+                location=OpenApiParameter.QUERY,
+                description="UUID of the processing job",
+            ),
+        ],
+        responses={
+            200: OpenApiResponse(
+                description="Processing job status",
+                response=PostProcessedDataSerializer,
+            ),
+            404: OpenApiResponse(description="Job not found"),
+        },
+    )
+    @action(detail=True, methods=["get"], url_path="spectrogram_status")
+    def get_spectrogram_status(self, request: Request, capture_uuid: str) -> Response:
+        """
+        Get the status of a spectrogram generation job.
+        """
+        if not settings.EXPERIMENTAL_SPECTROGRAM:
+            return Response(
+                {"error": "Spectrogram feature is not enabled"},
+                status=status.HTTP_403_FORBIDDEN,
+            )
+        try:
+            job_id = request.query_params.get("job_id")
+            if not job_id:
+                return Response(
+                    {"error": "job_id parameter is required"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            # Get the processing job
+            processing_job = get_object_or_404(
+                PostProcessedData,
+                uuid=job_id,
+                capture__uuid=capture_uuid,
+                processing_type=ProcessingType.Spectrogram.value,
+            )
+
+            serializer = PostProcessedDataSerializer(processing_job)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        except Exception as e:  # noqa: BLE001
```

Comment:
maybe this can be a `ValidationError` from `from rest_framework.serializers import ValidationError`; also try-except scope can be reduced